packages/common/src/hardhat/tasks/xchainSetup.ts

```diff
@@ -17,11 +20,103 @@ async function setupHub(hub: Contract, deployer: string, parentMessenger: string
   }
 }
 
+async function setupParentMessenger(
+  messenger: Contract,
+  deployer: string,
+  childMessenger: Deployment,
+  oracleHub: Deployment,
+  governorHub: Deployment,
+  oracleSpoke: Deployment,
+  governorSpoke: Deployment
+) {
+  const contractState = await Promise.all([
+    messenger.methods.getL1CallValue().call(),
+    messenger.methods.owner().call(),
+    messenger.methods.childMessenger().call(),
+    messenger.methods.oracleHub().call(),
+    messenger.methods.oracleSpoke().call(),
+    messenger.methods.governorHub().call(),
+    messenger.methods.governorSpoke().call(),
+  ]);
+  const messengerOwner = contractState[1];
+  const messengerChildMessenger = contractState[2];
+  const messengerOracleHub = contractState[3];
+  const messengerOracleSpoke = contractState[4];
+  const messengerGovernorHub = contractState[5];
+  const messengerGovernorSpoke = contractState[6];
+
+  // Submit parent messenger local transactions:
+  assert(
+    messengerOwner === deployer,
+    `Accounts[0] (${deployer}) is not equal to parent messenger owner (${messengerOwner})`
+  );
+  if (messengerChildMessenger !== childMessenger.address) {
+    console.log(`Setting child messenger to ${childMessenger.address}...`);
+    const setChildMessengerTxn = await messenger.methods
+      .setChildMessenger(childMessenger.address)
+      .send({ from: deployer });
+    console.log(`...txn: ${setChildMessengerTxn.transactionHash}`);
+  }
+  if (messengerOracleHub !== oracleHub.address) {
```

Comment:
I really like that if everything is already set up that this script just does nothing. 💯 very useful if something goes wrong halfway through, etc.